packages/toolkit/src/extension.ts

```diff
@@ -55,58 +53,31 @@ import { AwsResourceManager } from './dynamicResources/awsResourceManager'
 import { Experiments, Settings } from './shared/settings'
 import { isReleaseVersion } from './shared/vscode/env'
-import { Commands, registerErrorHandler as registerCommandErrorHandler } from './shared/vscode/commands2'
-import { UriHandler } from './shared/vscode/uriHandler'
 import { telemetry } from './shared/telemetry/telemetry'
 import { Auth } from './auth/auth'
-import { isUserCancelledError, resolveErrorMessageToDisplay, ToolkitError } from './shared/errors'
-import { Logging } from './shared/logger/commands'
-import { showMessageWithUrl, showViewLogsMessage } from './shared/utilities/messages'
-import { registerWebviewErrorHandler } from './webviews/server'
-import { ChildProcess } from './shared/utilities/childProcess'
+import { showViewLogsMessage } from './shared/utilities/messages'
 import { initializeNetworkAgent } from './codewhisperer/client/agent'
 import { Timeout } from './shared/utilities/timeoutUtils'
 import { submitFeedback } from './feedback/vue/submitFeedback'
-import { showQuickStartWebview } from './shared/extensionStartup'
-import { activateShared } from './extensionShared'
+import { activateShared, deactivateShared } from './extensionShared'
 
 let localize: nls.LocalizeFunc
 
+/**
+ * The entrypoint for the nodejs version of the toolkit
+ *
+ * **CONTIBUTORS** If you are adding code to this function prioritize adding it to
+ * {@link activateShared} if appropriate
+ */
 export async function activate(context: vscode.ExtensionContext) {
     const activationStartedOn = Date.now()
     localize = nls.loadMessageBundle()
 
-    globals.machineId = await getMachineId()
-
-    const remoteInvokeOutputChannel = vscode.window.createOutputChannel(
-        localize('AWS.channel.aws.remoteInvoke', '{0} Remote Invocations', getIdeProperties().company)
-    )
-
-    registerCommandErrorHandler((info, error) => {
-        const defaultMessage = localize('AWS.generic.message.error', 'Failed to run command: {0}', info.id)
-        void logAndShowError(error, info.id, defaultMessage)
-    })
-
-    registerWebviewErrorHandler((error: unknown, webviewId: string, command: string) => {
-        logAndShowWebviewError(error, webviewId, command)
-    })
-
-    if (isCloud9()) {
-        vscode.window.withProgress = wrapWithProgressForCloud9(globals.outputChannel)
-        context.subscriptions.push(
-            Commands.register('aws.quickStart', async () => {
-                try {
-                    await showQuickStartWebview(context)
-                } finally {
-                    telemetry.aws_helpQuickstart.emit({ result: 'Succeeded' })
-                }
-            })
-        )
-    }
-
     try {
         // IMPORTANT: If you are doing setup that should also work in browser, it should be done in the function below
-        await activateShared(context, () => RegionProvider.fromEndpointsProvider(makeEndpointsProvider()))
+        const extContext = await activateShared(context, () =>
```

Comment:
If it makes things easier, we could get rid of `ExtContext` . That is in the backlog

Reply:
This wasn't a pain yet, so will keep it as-is. Good to know we can drop it though since it is a bit awkward